src/script.rs

```diff
@@ -7,26 +7,23 @@ use crate::binding;
 use crate::bootstrap;
 use crate::js_loading;
 
-pub fn run_js_in_scope(scope: &mut v8::HandleScope, js: &str, filepath: &str) -> String {
+pub(crate) fn run_js_in_scope_internal<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    js: &str,
+    filepath: &str,
+) -> Option<v8::Local<'s, v8::Value>> {
     let filepath = v8::String::new(scope, filepath).unwrap();
     let origin = js_loading::create_script_origin(scope, filepath, false);
 
     let code = v8::String::new(scope, js).unwrap();
 
-    let tc_scope = &mut v8::TryCatch::new(scope);
-    let script = v8::Script::compile(tc_scope, code, Some(&origin));
-
-    if script.is_none() {
-        let exception = tc_scope.exception().unwrap();
-        let msg = v8::Exception::create_message(tc_scope, exception);
-        let error_message = msg.get(tc_scope).to_rust_string_lossy(tc_scope);
-        eprintln!("{}", &error_message);
-        return "".to_string();
-    }
+    v8::Script::compile(scope, code, Some(&origin)).and_then(|script| script.run(scope))
+}
 
-    let script = script.unwrap();
+pub fn run_js_in_scope(scope: &mut v8::HandleScope, js: &str, filepath: &str) -> String {
```

Comment:
this public api requires a HandleScope to run, I don't believe we have API that currently returns one... or am I mistaken.